src/Sulu/Bundle/MediaBundle/Entity/Media.php

```diff
@@ -172,6 +172,13 @@ public function setCollection(\Sulu\Bundle\MediaBundle\Entity\CollectionInterfac
      */
     public function getCollection()
     {
+        if (null === $this->collection) {
+            throw new \RuntimeException(sprintf(
+                'Media entity %d is not associated with a collection',
+                $this->getId()
+            ));
+        }
```

Comment:
Because it is NOT the case if you use Sqlite and delete the collection. The media exists and it has no collection. In anycase anything that should ALWAYS return an object should throw an exception if it is an invalid state (otherwise you get a Fatal Error).

I am assuming that no media entity should exist without a collection.